src/Simplex/Chat/Controller.hs

```diff
@@ -1044,6 +1050,27 @@ chatWriteVar :: ChatMonad' m => (ChatController -> TVar a) -> a -> m ()
 chatWriteVar f value = asks f >>= atomically . (`writeTVar` value)
 {-# INLINE chatWriteVar #-}
 
+chatModifyVar :: ChatMonad' m => (ChatController -> TVar a) -> (a -> a) -> m ()
+chatModifyVar f newValue = asks f >>= atomically . (`modifyTVar'` newValue)
+{-# INLINE chatModifyVar #-}
+
+setContactNetworkStatus :: ChatMonad' m => Contact -> NetworkStatus -> m ()
+setContactNetworkStatus ct = chatModifyVar connNetworkStatuses . M.insert (contactAgentConnId ct)
+
+-- iOS:
+-- keep <- + CRContactConnected: setContactNetworkStatus(contact, .connected)
+-- CRNetworkStatuses <- + CRContactSubSummary: setContactNetworkStatus(sub.contact, .connected) / processContactSubError(sub.contact, err)
+-- CRNetworkStatuses <- + CRContactSubError: processContactSubError(contact, chatError)
+-- keep <- + CRConnectedToGroupMember: m.setContactNetworkStatus(contact, .connected)
+-- CRNetworkStatus <- + CRContactsSubscribed: updateContactsStatus(contactRefs, status: .connected)
+-- CRNetworkStatus <- + CRContactsDisconnected: updateContactsStatus(contactRefs, status: .disconnected)
+
+-- + updateContactsStatus(): networkStatuses[c.agentConnId] = status
+-- + processContactSubError(): m.setContactNetworkStatus(contact, .error(err))
+
+-- Below seems pre-mature? maybe it was not connected yet? It will send contact connected and it will set the status, that will be spinning until then?
```

Comment:
possibly.. it may seem counterintuitive that user should send message when there's spinner

Reply:
possibly